engine/src/main/java/io/zeebe/engine/state/appliers/ErrorThrownEventApplier.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.state.appliers;
+
+import io.zeebe.engine.processing.common.ErrorEventHandler;
+import io.zeebe.engine.state.TypedEventApplier;
+import io.zeebe.engine.state.ZeebeState;
+import io.zeebe.engine.state.mutable.MutableElementInstanceState;
+import io.zeebe.engine.state.mutable.MutableJobState;
+import io.zeebe.protocol.impl.record.value.job.JobRecord;
+import io.zeebe.protocol.record.intent.JobIntent;
+
+public class ErrorThrownEventApplier implements TypedEventApplier<JobIntent, JobRecord> {
+
+  private final MutableJobState jobState;
+  private final MutableElementInstanceState elementInstanceState;
+  private final ErrorEventHandler errorEventHandler;
+
+  ErrorThrownEventApplier(final ZeebeState state) {
+    jobState = state.getJobState();
+    elementInstanceState = state.getElementInstanceState();
+
+    errorEventHandler =
+        new ErrorEventHandler(
+            state.getWorkflowState(),
+            state.getElementInstanceState(),
+            state.getEventScopeInstanceState(),
+            state.getKeyGenerator());
```

Comment:
We cannot put it into the state, because the information can only be derived by looking at workflow state and element instance state combined.
We cannot put it in the applier, because the processor wants to call some of the same logic.

I do see your point. 

I refactored this and put it into a new class which I call an analyzer. This is a new category of class. It takes only the read only state. but can look at several readonly state objects combined. The signature is slimmed down.

Reply:
After discussing it with Phil he proposed a different approach:
* reject the throw error command if no catch event can be found
* Move trigger event logic into event applier
* Extend tests to cover these cases